Make options available in all Blueprint hooks as this.options

trentmwillis opened the pull request:
Currently, not all the available Blueprint hooks receive the `options` hash (e.g., `filesPath` and `normalizeEntityName`). This is a simple change to ensure that all hooks can access the options via `this.options`.

cc @rwjblue

rwjblue commented:
I'm :+1: on passing in `options`.  Can you add a test that confirms `this.options` is present in `files` hook (or another hook if there are easier ones)?

trentmwillis commented:
Will add a test. Would you want me to pass `this.options` into `filesPath` as an argument (e.g., `this.filesPath(this.options)`)? Or should just having `this.options` be fine?

trentmwillis commented:
Added tests. I noticed that most of the blueprint hooks didn't really have any tests, so I just added some simple ones to make sure they get called with the proper arguments during install/uninstall.

Also documented `filesPath` as one of the available hooks and set it up so that it got `options` as an argument (similar to the other hooks).

rwjblue commented:
Awesome, thank you!
